packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -55,7 +55,6 @@ const DEFAULT_CSP = {
   'script-src': ["'self'", "'unsafe-eval'"],
   'script-src-attr': ["'none'"],
   'style-src': ["'self'", 'https:', "'unsafe-inline'"],
-  'upgrade-insecure-requests': [] as string[],
```

Comment:
wouldn't it be best to have this enabled only essentially if `app.baseUrl` starts with `https:`?

Reply:
From the [CSP docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/upgrade-insecure-requests):

> The HTTP Content-Security-Policy (CSP) upgrade-insecure-requests directive instructs user agents to treat all of a site's insecure URLs (those served over HTTP) as though they have been replaced with secure URLs (those served over HTTPS). This directive is intended for web sites with large numbers of insecure legacy URLs that need to be rewritten.

This concept doesn't really apply here, if the `baseUrl` is HTTPS then Backstage URLs will use that. We don't really need the layer on top unless you want to be truly belts-and-suspenders about it, in which case you can enable it. As mentioned, this is more intended for legacy websites where you can't reason intelligently about all the links.